autosklearn/pipeline/components/classification/mlp.py

```diff
@@ -0,0 +1,232 @@
+            self.num_nodes_per_layer = int(self.num_nodes_per_layer)
+            self.hidden_layer_sizes = tuple(self.num_nodes_per_layer
+                                            for i in range(self.hidden_layer_depth))
+            self.activation = str(self.activation)
+            self.alpha = float(self.alpha)
+            self.learning_rate_init = float(self.learning_rate_init)
+            self.early_stopping = str(self.early_stopping)
+            if self.early_stopping == "off":
+                self.early_stopping_val = False
+                # self.validation_fraction = 0.0
+                # self.n_iter_no_change = int(self.n_iter_no_change)
+                # self.tol = float(self.tol)
+            else:
+                raise ValueError("Can't use early_stopping (set to %s)" % self.early_stopping)
+                # self.early_stopping_val = True
+                # self.validation_fraction = 0.0
+                # self.n_iter_no_change = self.max_iter
+                # self.tol = 0
+            self.solver = self.solver
+
+            try:
+                self.batch_size = int(self.batch_size)
+            except ValueError:
+                self.batch_size = str(self.batch_size)
+
+            self.shuffle = check_for_bool(self.shuffle)
+            self.beta_1 = float(self.beta_1)
+            self.beta_2 = float(self.beta_2)
+            self.epsilon = float(self.epsilon)
+            self.beta_1 = float(self.beta_1)
+
+            self.max_iter = int(self.max_iter)
+            self.verbose = int(self.verbose)
+
+            n_iter = int(np.ceil(n_iter))
+
+            # initial fit of only increment trees
+            self.estimator = MLPClassifier(
+                hidden_layer_sizes=self.hidden_layer_sizes,
+                activation=self.activation,
+                solver=self.solver,
+                alpha=self.alpha,
+                batch_size=self.batch_size,
+                learning_rate_init=self.learning_rate_init,
+                max_iter=n_iter,
+                shuffle=self.shuffle,
+                random_state=self.random_state,
+                verbose=self.verbose,
+                warm_start=True,
+                early_stopping=self.early_stopping_val,
+                # validation_fraction=self.validation_fraction,
+                # n_iter_no_change=self.n_iter_no_change,
+                # tol=self.tol,
+                beta_1=self.beta_2,
+                beta_2=self.beta_1,
+                epsilon=self.epsilon,
+                # We do not use these, see comments below in search space
+                # momentum=self.momentum,
+                # nesterovs_momentum=self.nesterovs_momentum,
+                # power_t=self.power_t,
+                # learning_rate=self.learning_rate,
+                # max_fun=self.max_fun
+            )
+            self.estimator.fit(X, y)
+        else:
+            # **NOTE** This is fixed in scikit-learn > 0.23.2
```

Comment:
Could you please also note what needs to be changed when we go to scikit-learn 0.23.3, maybe with a TODO?

Reply:
Sure